packages/model-viewer/src/features/scene-graph/image.ts

```diff
@@ -52,37 +51,64 @@ export class Image extends ThreeDOMElement implements ImageInterface {
   }
 
   get name(): string {
-    return (this[$sourceObject] as any).name || '';
+    return (this[$sourceObject] as GLTFImage).name || '';
   }
 
   get uri(): string|undefined {
-    return this[$uri];
+    return (this[$sourceObject] as GLTFImage).uri;
+  }
+
+  get bufferView(): number|undefined {
+    return (this[$sourceObject] as GLTFImage).bufferView;
   }
 
   get type(): 'embedded'|'external' {
     return this.uri != null ? 'external' : 'embedded';
   }
 
   async setURI(uri: string): Promise<void> {
-    this[$uri] = uri;
+    (this[$sourceObject] as GLTFImage).uri = uri;
 
     const image = await new Promise((resolve, reject) => {
       loader.load(uri, resolve, undefined, reject);
     });
 
-    const texture = this[$textureInfo][$threeTexture];
-    if (texture) {
-      // If the URI is set to null but the Image had an associated buffer view
-      // (this would happen if it started out as embedded), then fall back to
-      // the cached object URL created by GLTFLoader:
-      if (image == null &&
-          (this[$sourceObject] as GLTFEmbeddedImage).bufferView != null) {
-        texture.image = this[$bufferViewImages].get(texture);
```

Comment:
I removed the bufferViewImages because they don't really make any sense to me. It's like you can change the image and then *sometimes* you have a previous image to go back to? Nah, and the simplification will probably be necessary as we expand this material API.